### PR TITLE
refactor(explore): Enhance Dataset and Control panel Collapse components

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Removes react-bootstrap Collapse with Antdesign enhanced Collapse.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/60598000/105727326-41f8f800-5f2b-11eb-8ae3-bf7ee4d7e866.mp4

After:

https://user-images.githubusercontent.com/60598000/105727342-46bdac00-5f2b-11eb-9826-dae94ae274dd.mp4

### TEST PLAN
1. Open the explorer
2. Make sure the control panel has all the collapsible expanded
3. Close a collapsible panel, the content should be hidden
4. Re-open the closed collapsible panel, the content should appear

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by geido** [comment]
@rusackas @junlincc

**Comment by junlincc** [comment]
thanks @geido for the PR. if the caret was the only thing that changed, it LGTM. but im not sure why the tabs in the control panel got messed up a bit... 
Screen Shot 2021-01-01 at 2 10 13 PM
this is what im seeing and how it supposed to look in another open PR
Screen Shot 2021-01-01 at 2 11 47 PM

**Comment by geido** [comment]
> thanks @geido for the PR. if the caret was the only thing that changed, it LGTM. but im not sure why the tabs in the control panel got messed up a bit... 
> Screen Shot 2021-01-01 at 2 10 13 PM
> this is what im seeing and how it supposed to look in another open PR
> 
> 
> Screen Shot 2021-01-01 at 2 11 47 PM
> 

Thanks @junlincc. Let me check what might have caused that.

**Comment by geido** [comment]
@junlincc Please check now. The issue should be resolved.

**Review by junlincc — APPROVED** [review]
manually tested, LGTM!

If it's easy, please remove 'Dataset' from the Dataset & Chart Type Section in this PR as well. 
Screen Shot 2021-01-04 at 9 07 55 PM

and get the tests passed.

**Comment by ktmud** [comment]
The caret directions are inconsistent with the datasource panel.

**Folded:**
image

**Expanded:**

image


I think the directions used by the datasource pane makes more sense.

**Comment by ktmud** [comment]
Also, can we fix the font-size inconsistency in these panes as well (shouldn't they use the same styled component)?

**Comment by ktmud** [comment]
May I also suggest tune the font size/spacing of the columns/metrics as well?

**Before**:


**After**:

(a rough adjustment, could be better)

**Comment by ktmud** [comment]
The title border spacing is also inconsistent.

**Comment by geido** [comment]
Hello @ktmud. Thanks for your feedback. We are working on moving the Collapse components to Antd in small steps. This is the reason why you see inconsistencies between the two. We should use Antdesign as a reference though.

Concerning the fonts, react-bootstrap was having a 16px overriding the default 14px. This should be an improvement. Unless you think keeping the 16px imposed by react-bootstrap was intentional.

As for the direction of the arrows, those are by default in Antd. While I agree with you that the directions used by the previous Collapse are more intuitive, this is a change that will have an impact on all the Collapse components already moved to Antd. I would keep it as an item for later.

**Comment by ktmud** [comment]
Cool. Glad to hear this is an ongoing migration. Just documenting what I found here in case they slip through.

**Comment by geido** [comment]
After another look, I have noticed that the DatasourcePanel is using the Antd Collapse already, just not the enhanced one that we have introduced in the common components, but another flavor of it. This will need to be consolidated. Most probably all the existing instances of the Antd Collapse that aren't using the enhanced one will need to be moved to the enhanced.  Thoughts @rusackas?

**Comment by rusackas** [comment]
After much discussion with the Preset design team (other voices can pitch in here!) we think we'd like to keep the current arrow directions. 

As far as arrow **direction** goes, the design proposal is this: If the collapse arrow is on the right side of the content area (like controls), it will point up when open or down when closed.. If it's on the left side of a header (like the Data pane) it will point right when closed, and down when open.

The above is more or less in accordance with the Apple [guidelines](https://developer.apple.com/design/human-interface-guidelines/macos/buttons/disclosure-controls/).

We should utilize a prop in the Collapse component to pick which version we're using... it appears the `expandIconPosition` could trigger conditional styling.

There's also a discrepancy in **animation** - we agreed that they should all be animated (and quickly!) with clockwise rotation when opening, and counterclockwise when closing.

And in coming to all these conclusions, we came up with a whole system of floating design proposals as Github Issues, then moving them to the Github wiki when consensus is reached. We'll be sending a proposal around this to the `dev@` list soon!

**Comment by ktmud** [comment]
While we are here, can we also fix the click area?

Currently the Dateset panel accordions can click to fold/expand on the whole header section, but the Controls panel accordions only have the text and the caret clickable.

**Comment by ktmud** [comment]
Personally I'm also not a fan of caret animation (and even the slide up/slide down animation maybe we can also get rid of). 

Airbnb design guidelines for internal tools is also avoid animation as much as possible. We don't even have animation for modals and popovers.

For tools people will use everyday, (the presumed) efficiency and speed are probably more important than the "pleasant surprises" animations are normally designed to inspire.

**Comment by geido** [comment]
> While we are here, can we also fix the click area?

@ktmud this PR will bring that for free as the Antdesign Collapse has the full header area clickable

**Comment by geido** [comment]
@rusackas @ktmud I have made some adjustments so that the Dataset panel and Control panels use the same styling. As for the open points, concerning animation and direction of the arrows, let's come to a consensus so that I can close this up.

I tend to agree to avoid animations. As for the direction of the arrows, I am fine with what Evan has proposed.

**Comment by geido** [comment]
I have made more changes as follows:

- Added an `animateArrows` prop so that those animations can be optional
- Implemented the arrow open/close position based on whether the `expandIconPosition` is right or left
- Added a storybook entry to show the arrows when the animation is ON
- I have enhanced also the DataSource panel, so it looks the same as the Control panel
- I have also updated the description with the updated videos showing the functionality

This is quite a big change. I will appreciate support with testing eventual edge cases.

Thank you!

@junlincc @rusackas @ktmud

File: superset-frontend/src/common/components/Collapse.tsx [hunk 1]
```diff
@@ -68,6 +90,18 @@ const Collapse = Object.assign(
...
       }
     }
+    .ant-collapse-item-active {
+      .ant-collapse-header {
+        ${({ expandIconPosition }) =>
+          expandIconPosition &&
+          expandIconPosition === 'right' &&
+          `
+            .anticon.anticon-right.ant-collapse-arrow > svg {
+              transform: rotate(-90deg) !important;
```
Comment (maloun96) [inline superset-frontend/src/common/components/Collapse.tsx:100]: can we avoid using important?

Comment (geido) [reply superset-frontend/src/common/components/Collapse.tsx:100]: Nope. This is added by the library on the element itself, which makes the important necessary in order to override

**Review by ktmud — APPROVED** [review]
LGTM. Hope we can have a `Collapse.test.tsx` file but that's a low priority.

**Comment by adam-stasiak** [comment]
I see that when I click on second collapse then field with ```search metrics and columns``` jumps a little bit.
Are we able to fix this?

https://user-images.githubusercontent.com/25153919/105753898-ad51c280-5f49-11eb-8117-61a8d4b6fb7e.mov

**Comment by geido** [comment]
Hello @adam-stasiak I am seeing the same problem on master. Would you like to create a separate issue for that?

https://www.awesomescreenshot.com/video/2508610?key=292778fadfc41d2f0f20f3099117aea6

Thanks

**Comment by adam-stasiak** [comment]
ah ok! I will create! sorry for inconvenience.

**Comment by adam-stasiak** [comment]
after manual testing looks fine to me 🟢

**Pull request merged**